packages/amesos2/src/Amesos2_Superlu_TypeMap.hpp

```diff
@@ -211,48 +254,93 @@ class ValueTypeConversionTraits<SLU::Z::doublecomplex, TypeFrom>
 
 
 // Also convert from SLU types
-template <typename TypeTo>
-class ValueTypeConversionTraits<TypeTo, SLU::C::complex>
+
+template <>
+class ValueTypeConversionTraits<std::complex<float>, SLU::C::complex>
+{
+public:
+  static std::complex<float> convert( const SLU::C::complex t )
+    {
+      typedef typename std::complex<float>::value_type value_type;
+      value_type ret_r = Teuchos::as<value_type>( t.r );
+      value_type ret_i = Teuchos::as<value_type>( t.i );
+      return ( std::complex<float>( ret_r, ret_i ) );
+    }
+
+  // No special checks for safe Convert
+  static std::complex<float> safeConvert( const SLU::C::complex t )
+    {
+      typedef typename std::complex<float>::value_type value_type;
+      value_type ret_r = Teuchos::as<value_type>( t.r );
+      value_type ret_i = Teuchos::as<value_type>( t.i );
+      return ( std::complex<float>( ret_r, ret_i ) );
+    }
+};
+
+template <>
+class ValueTypeConversionTraits<std::complex<double>, SLU::C::complex>
 {
 public:
-  static TypeTo convert( const SLU::C::complex t )
+  static std::complex<double> convert( const SLU::C::complex t )
     {
-      typedef typename TypeTo::value_type value_type;
+      typedef typename std::complex<double>::value_type value_type;
       value_type ret_r = Teuchos::as<value_type>( t.r );
       value_type ret_i = Teuchos::as<value_type>( t.i );
-      return ( TypeTo( ret_r, ret_i ) );
+      return ( std::complex<double>( ret_r, ret_i ) );
     }
 
   // No special checks for safe Convert
-  static TypeTo safeConvert( const SLU::C::complex t )
+  static std::complex<double> safeConvert( const SLU::C::complex t )
     {
-      typedef typename TypeTo::value_type value_type;
+      typedef typename std::complex<double>::value_type value_type;
       value_type ret_r = Teuchos::as<value_type>( t.r );
       value_type ret_i = Teuchos::as<value_type>( t.i );
-      return ( TypeTo( ret_r, ret_i ) );
+      return ( std::complex<double>( ret_r, ret_i ) );
     }
 };
 
 
-template <typename TypeTo>
-class ValueTypeConversionTraits<TypeTo, SLU::Z::doublecomplex>
+template <>
+class ValueTypeConversionTraits<std::complex<float>, SLU::Z::doublecomplex>
+{
+public:
+  static std::complex<float> convert( const SLU::Z::doublecomplex t )
+    {
+      typedef typename std::complex<float>::value_type value_type;
+      value_type ret_r = Teuchos::as<value_type>( t.r );
+      value_type ret_i = Teuchos::as<value_type>( t.i );
+      return ( std::complex<float>( ret_r, ret_i ) );
```

Comment:
Tip for the future: this works too:
```c++
return { ret_r, ret_i };
```

Reply:
@mhoemmen thanks for pointing that out, is that preferred for returning complex types, as in style or performance benefits?

Reply:
@ndellingwood It's just more concise.